Client/FileManagerExtension.swift

```diff
@@ -40,4 +41,50 @@ public extension FileManager {
         }
         return false
     }
+    
+    func getOrCreateDirectory(withName name: String) -> (path: String, created: Bool) {
+        guard let documentDirectory = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first else {
+            log.error("Can't get documents dir.")
+            return ("", false)
+        }
+        
+        let path = documentDirectory + "/" + name
+        var wasCreated = false
+        if !fileExists(atPath: path) {
+            do {
+                try createDirectory(atPath: path, withIntermediateDirectories: false, attributes: nil)
+            } catch {
+                log.error("createDirectory error: \(error)")
+            }
+            wasCreated = true
+        }
+        return (path, wasCreated)
+    }
+    
+    func writeToDiskInFolder(_ data: Data, fileName: String, folderName: String) -> Deferred<()> {
+        let completion = Deferred<()>()
```

Comment:
No `Deferred` is needed here since this function is serial. Just have no return type.